### PR TITLE
PersonaButton, PersonaButtonCarousel, and CardNudge from vNext

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This merges two new controls from the `vnext-prototype` branch: `PersonaButton` and `PersonaButtonCarousel`. These supersede the legacy `ContactCollectionView` and `ContactCollectionViewCell`.

Additionally, this change upgrades `CardNudge` to utilize the full vNext-style design tokens system. All behaviors, appearance, and APIs are unchanged.

### Verification

Ensured that `PersonaButton`, `PersonaButtonCarousel`, and `CardNudge` behave identically as in the vNext branch, including:
- Color theming
- Dark/light mode
- Accessibility

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] VoiceOver and Keyboard Accessibility
- [x] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/676)

### CODE REVIEW

File: ios/FluentUI.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -151,6 +147,7 @@
 		5314E2F525F025C60099271A /* CalendarConfiguration.swift in Sources */ = {isa = PBXBuildFile; fileRef = FDF41ED82141A02200EC527C /* CalendarConfiguration.swift */; };
 		5314E30225F0260E0099271A /* AccessibilityContainerView.swift in Sources */ = {isa = PBXBuildFile; fileRef = FD77752A219E455A00033D58 /* AccessibilityContainerView.swift */; };
 		5314E30325F0260E0099271A /* AccessibleViewDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = FD599D052134A682008845EE /* AccessibleViewDelegate.swift */; };
+		5360994126B8B2710069DE71 /* PersonaButtonCarousel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 929DD258266ED3B600E8175E /* PersonaButtonCarousel.swift */; };
```
Comment (rdeassis) [inline ios/FluentUI.xcodeproj/project.pbxproj:150]: There needs to be a subspec for the PersonaButtonCarousel in the main podspec.

**Review by rdeassis — CHANGES_REQUESTED** [review]

File: ios/FluentUI/Vnext/PersonaButton/MSFPersonaButton.swift [hunk 2]
```diff
@@ -0,0 +1,167 @@
+//
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Vnext/PersonaButton/MSFPersonaButton.swift:1]: Ah dang I was sure I moved those, probably in a different failed local merge. Thanks!

File: ios/FluentUI/Vnext/PersonaButton/MSFPersonaButton.swift [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            return self.personaButton.state.onTapAction
+        }
+        set {
+            self.personaButton.state.onTapAction = newValue
+        }
+    }
+
+    // MARK: - MSFPersonaButtonData
+
+    @objc public var avatarBackgroundColor: UIColor? {
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Vnext/PersonaButton/MSFPersonaButton.swift:52]: Yeah I think this was a pretty cool experiment, but I'm on board with standardizing on state exposure. Same with your other comment: time to unify the protocols and expose something different on the carousel.

File: ios/FluentUI/Vnext/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift [hunk 4]
```diff
@@ -0,0 +1,105 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+/// UIKit wrapper that exposes the SwiftUI PersonaGrid implementation
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Vnext/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift:9]: aw dang ☹️

File: ios/FluentUI/Vnext/PersonaButtonCarousel/PersonaButtonCarousel.swift [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            }
+        }
+        .background(Color(tokens.backgroundColor))
+        .designTokens(tokens,
+                      from: theme,
+                      with: windowProvider)
+    }
+}
+
+/// Properties that make up PersonaGrid content
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Vnext/PersonaButtonCarousel/PersonaButtonCarousel.swift:51]: aw dang ☹️

**Review by harrieshin — APPROVED** [review]

File: ios/FluentUI/PersonaButton/MSFPersonaButton.swift [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    @objc open var view: UIView {
+        return hostingController.view
+    }
+
+    /// The object that groups properties that allow control over the PersonaButton appearance.
+    @objc open var state: MSFPersonaButtonState {
+        return personaButton.state
+    }
+
+    @objc public init(size: MSFPersonaButtonSize = .large,
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButton/MSFPersonaButton.swift:22]: nit: Do you mind adding documentation comments for the initializer here as well?

File: ios/FluentUI/PersonaButton/PersonaButton.swift [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+
+/// View that represents a persona button.
+public struct PersonaButton: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @Environment(\.sizeCategory) var sizeCategory: ContentSizeCategory
+    @ObservedObject var tokens: MSFPersonaButtonTokens
+    @ObservedObject var state: MSFPersonaButtonStateImpl
+
+    public init(size: MSFPersonaButtonSize) {
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButton/PersonaButton.swift:64]: nit: Do you mind adding documentation for this one too?

File: ios/FluentUI/PersonaButton/MSFPersonaButton.swift [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+                      theme: FluentUIStyle? = nil) {
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButton/MSFPersonaButton.swift:23]: As we discussed this morning, it's probably better to keep the convenience initializer that omits the theme parameter in the interest of making the objc calls less verbose.

File: ios/FluentUI/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+import UIKit
+import SwiftUI
+
+/// UIKit wrapper that exposes the SwiftUI PersonaButtonCarousel implementation
+@objc open class MSFPersonaButtonCarousel: NSObject, FluentUIWindowProvider {
+
+    // MARK: - Public API
+
+    @objc open var view: UIView {
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift:14]: nit: Do you mind adding documentation for this one too?

File: ios/FluentUI/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        return hostingController.view
+    }
+
+    @objc open var state: MSFPersonaCarouselState {
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift:18]: nit: Do you mind adding documentation for this one too?

File: ios/FluentUI/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        return self.personaButtonCarousel.state
+    }
+
+    @objc public init(size: MSFPersonaButtonSize = .large,
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift:22]: nit: Do you mind adding documentation for this one too?

File: ios/FluentUI/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+                      theme: FluentUIStyle? = nil) {
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift:23]: Same as the other comment about the convenience initializer to make objc calls less verbose.

File: ios/FluentUI/PersonaButtonCarousel/PersonaButtonCarousel.swift [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+    /// - Returns: An optional reference to the added `PersonaButton`, which can be used to set additional properties or to update later.
+    @discardableResult func add(primaryText: String?, secondaryText: String?, image: UIImage?) -> MSFPersonaCarouselButtonState
+
+    /// Retrieves the `PersonaButton` at a given index, or nil if the index is out of bounds.
+    ///
+    /// - Parameters:
+    ///   - index: The index of the `PersonaButton` to retrieve
+    ///
+    /// - Returns: A reference to the  `PersonaButton` at the given index if one exists.
+    func personaButtonData(at index: Int) -> MSFPersonaCarouselButtonState?
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButtonCarousel/PersonaButtonCarousel.swift:35]: nit: maybe renaming personaButtonData to personaButtonState?

File: ios/FluentUI/PersonaButtonCarousel/PersonaButtonCarousel.swift [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+
+    /// Removes a `PersonaButton` from the carousel.
+    ///
+    /// - Parameters:
+    ///   - personaData: The reference to a `PersonaButton` to be removed.
+    func remove(_ personaData: MSFPersonaCarouselButtonState)
```
Comment (rdeassis) [inline ios/FluentUI/PersonaButtonCarousel/PersonaButtonCarousel.swift:41]: nit: same here? maybe renaming the parameter to personaButtonState?

**Review by rdeassis — APPROVED** [review]

**Pull request merged**